File: Daily challenge/Find if Path Exists in Graph.py
class Solution:
    def validPath(self, n: int, edges: List[List[int]], source: int, destination: int) -> bool:
        if n == 1: return True
        vi = [False]*n
        vi[source] = True
        flag = True
        while flag:
            flag = False
            for edge in edges:
                if vi[edge[0]] != vi[edge[1]]:
                    vi[edge[0]] = True
                    vi[edge[1]] = True
                    flag = True
                
                if vi[destination]: return True
        
        return False

File: Daily challenge/Find if Path Exists in Graph.cpp
class Solution 
{
public:
    bool validPath(int n, vector<vector<int>>& edges, int source, int destination) 
    {
        if(n == 1) return true;
        vector<bool> vi(n, false);
        vi[source] = true;
        bool flag = true;
        while(flag)
        {
            flag = false;
            for(int i = 0; i < edges.size(); i++)
            {
                if(vi[edges[i][0]] != vi[edges[i][1]])
                {
                    vi[edges[i][0]] = true;
                    vi[edges[i][1]] = true;
                    flag = true;
                }
                if(vi[destination]) return true;
            }
        }
        return false;
    }
};